maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,215 @@
+   in https://issues.redhat.com/browse/WFLY-14936[WFLY-14936].
+
+=== Hard Requirements
+
+==== Plugin requirements
+
+* The Maven plugin produces inside the project target dir a directory (named ```server``` by default) that contains the server and any deployment.
+
+==== WildFly Maven Plugin new goals
+
+* ```wildfly:package```: Galleon provisioning of a server, CLI script execution, copy content to the server, deploy application (if present). 
+* ```wildfly:provision```: Galleon provisioning of a server.
+
+In addition to these 2 new goals, we are replacing the existing automatic download and unzip of wildfly-dist by a Galleon provisioning of 
+WildFly. This change impacts ```run``` and ```start```goals).
+ 
+==== Galleon related Plugin configuration items
+
+These configuration items are common to the ```provision``` and ```package``` goals. They are used to configure 
+Galleon to provision a server.
+
+* feature-packs: A list of feature-pack to install.
+* log-provisioning-time: Whether to log provisioning time at the end.
+* offline-provisioning: Whether to use offline mode when the plugin resolves an artifact.
+* galleon-options: Arbitrary Galleon options used when provisioning the server.
+* provisioning-dir: The name of the directory relative to the Maven project build directory used when the server is provisioned. 
+It defaults to ``server`` and will result in the server being provisioned in ``${project.build.directory}/server`` (which is resolved to ``target/server`` by default)
+* provisioning-file: The path to the provisioning.xml file to use, by default ```<project base dir>/galleon/provisioning.xml``` is used. If some feature-packs have been set, 
+the provisioning file is being ignored.
+* record-provisioning-state: Whether to record provisioned state in .galleon directory. When false, the file ```.wildfly-maven-plugin-provisioning.xml``` is generated in the server home directory.
+It contains Galleon provisioning information used to provision this server.
+* layers: A set of layers to include when building-up a custom configuration.
+* excluded-layers: A set of layers to exclude when building-up a custom configuration.
+
+==== ```provision``` goal specific behavior
+
+Calling ```provision``` goal without any Galleon configuration items (no feature-pack provided nor provisioning.xml file) 
+will provision the same default server that we are provisioning when calling ```run``` and ```start``` goals.
+
+==== ```package``` goal specific configuration items and behavior
+
+A Galleon configuration is expected, if no galleon configuration is found, the goal execution fails.
+
+The configuration items names, when applicable, are re-using the names that are in use in existing WildFly Maven plugin goals.
+
+* packaging-scripts: A list of execution of CLI scripts and commands. Each execution can contain: a list of CLI scripts, a list of commands, a list of properties files, 
```

Comment:
What is the difference between `a list of CLI scripts` and `a list of commands`? In what combinations these can be used?

Reply:
User as freedom to inline commands in the pom and/or reference CLI scripts. They can choose their own granularity for scripts vs commands.